### PR TITLE
Add picard/sortvcf

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

Adds sortvcf from picard

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/picard/sortvcf/main.nf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    picard \\
+        SortVcf \\
```
Comment (sateeshperi) [inline modules/picard/sortvcf/main.nf:25]: does sortvcf need any setting for mem as in other picard tools ? ([Example](https://github.com/nf-core/modules/blob/fba1b381f5629d2dfce3ec2350dc03821fd275ad/modules/picard/mergesamfiles/main.nf#L33))

File: modules/picard/sortvcf/meta.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: picard_sortvcf
+description: Sorts vcf files
+keywords:
+  - sort
+  - vcf
+tools:
+  - picard:
+      description: Java tools for working with NGS data in the BAM/CRAM/SAM and VCF format
+      homepage: https://broadinstitute.github.io/picard/
+      documentation: https://broadinstitute.github.io/picard/
```
Comment (sateeshperi) [inline modules/picard/sortvcf/meta.yml:10]: recommend this specific [link](https://broadinstitute.github.io/picard/command-line-overview.html#SortVcf) for documentation

File: modules/picard/sortvcf/main.nf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        --INPUT $vcf \\
+        --OUTPUT ${prefix}_sorted.vcf
```
Comment (sateeshperi) [inline modules/picard/sortvcf/main.nf:27]: is there an option for compressing the output vcf file?

File: modules/picard/sortvcf/main.nf (hunk 1 above)
Comment (ramprasadn) [reply modules/picard/sortvcf/main.nf:34]: Good point! I have added that option now.

File: modules/picard/sortvcf/meta.yml (hunk 2 above)
Comment (ramprasadn) [reply modules/picard/sortvcf/meta.yml]: Fixed 👍🏻

File: modules/picard/sortvcf/main.nf (hunk 3 above)
Comment (ramprasadn) [reply modules/picard/sortvcf/main.nf:38]: No, this tool doesn't have that option.

File: tests/modules/picard/sortvcf/test.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+## TODO nf-core: Please run the following command to build this file:
```
Comment (sateeshperi) [inline tests/modules/picard/sortvcf/test.yml:1]: remove TODO

File: tests/modules/picard/sortvcf/test.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+- name: picard sortvcf
+  command: nextflow run ./tests/modules/picard/sortvcf -entry test_picard_sortvcf -c ./tests/config/nextflow.config -c ./tests/modules/picard/sortvcf/nextflow.config
+  tags:
+    - picard
+    - picard/sortvcf
+  files:
+    - path: output/picard/test_sorted.vcf
```
Comment (sateeshperi) [inline tests/modules/picard/sortvcf/test.yml:7]: was the md5sum variable that you removed it? @ramprasadn

File: modules/picard/sortvcf/main.nf (hunk 3 above)
Comment (mahesh-panchal) [reply modules/picard/sortvcf/main.nf:38]: You can use `gzip` in the container/env to compress the output.

File: modules/picard/sortvcf/main.nf [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    picard \\
+        SortVcf \\
+        -Xmx${avail_mem}g \\
+        --INPUT $vcf \\
+        $seq_dict \\
+        --OUTPUT ${prefix}_sorted.vcf
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        picard: \$(picard SortVcf --version 2>&1 | grep -o 'Version:.*' | cut -f2- -d:)
```
Comment (mahesh-panchal) [inline modules/picard/sortvcf/main.nf:42]:
```suggestion
        picard: \$( picard SortVcf --version |& sed '2!d; s/Version://' )
```

File: modules/picard/sortvcf/main.nf [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    output:
+    tuple val(meta), path("*_sorted.vcf"), emit: vcf
+    path "versions.yml"                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
```
Comment (mahesh-panchal) [inline modules/picard/sortvcf/main.nf:22]: Does `picard Sortvcf` take `$args`?

Comment (ramprasadn) [reply modules/picard/sortvcf/main.nf:22]: Good suggestion! Will fix it 👍🏻

Comment (ramprasadn) [reply modules/picard/sortvcf/main.nf:22]: Turns out `--CREATE_MD5_FILE <Boolean>`  only works for bam and fastq files, and not vcfs. But I have added an optional input for `--REFERENCE_SEQUENCE` now.

File: modules/picard/sortvcf/main.nf (hunk 3 above)
Comment (ramprasadn) [reply modules/picard/sortvcf/main.nf:38]: Fixed it 👍🏻 All I had to do was change the output file's suffix from .vcf to .vcf.gz 🤷🏻‍♂️ 🤦🏽

**Review by sateeshperi — APPROVED** [review]
lgtm! Thanks for working with the reviews and updates

**Comment by mahesh-panchal** [comment]
What was the conda error that was fixed?

**Comment by sateeshperi** [comment]
i remember it was the `versions.yml` not being captured properly

**Comment by mahesh-panchal** [comment]
I saw the version capture changed back so I thought that was the issue.
If it's something that is platform dependent, it would be nice to know where it's not working.

**Pull request merged**